sql/core/src/test/resources/sql-tests/results/postgreSQL/union.sql.out

```diff
@@ -684,8 +684,8 @@ SELECT cast('3.4' as decimal(38, 18)) UNION SELECT 'foo'
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkException
-Failed to merge incompatible data types decimal(38,18) and string
+org.apache.spark.sql.AnalysisException
+Union can only be performed on tables with the compatible column types. string <> decimal(38,18) at the first column of the second table
```

Comment:
`string <> decimal(38,18) at the first column of the second table` is it valid English syntax?

Reply:
Ha, this comes from `CheckAnalysis`'s original error message. We can improve it, although there are some more tests relying on the error message.

Reply:
Updated the error message. Please let me know if it looks good to you.